### PR TITLE
Added openai_schema decorator

### LINKED ISSUE #17: Change openai base schema to a decorator so models only inherit from base model.

### PR DESCRIPTION
**Pull request by DaveOkpare**
Fixes #17 

Added `openai_schema` decorator that inherits from BaseModel

```python3
@openai_schema
class MyClass(BaseModel):
    ...

```

### CODE REVIEW

**Comment by jxnl** [comment]
Mind adding some tests

File: openai_function_call/function_calls.py [hunk 1]
```diff
@@ -109,3 +109,43 @@ def from_response(cls, completion, throw_error=True):
...
+                assert "function_call" in message, "No function call detected"
+                assert (
+                    message["function_call"]["name"] == cls.openai_schema["name"]
+                ), "Function name does not match"
+
+            function_call = message["function_call"]
+            arguments = json.loads(function_call["arguments"])
+            return cls(**arguments)
+
+    Wrapper.__name__ = cls.__name__
```
Comment (jxnl) [inline openai_function_call/function_calls.py:148]: Use functools wraps

File: openai_function_call/function_calls.py [hunk 2]
```diff
@@ -109,3 +109,43 @@ def from_response(cls, completion, throw_error=True):
         function_call = message["function_call"]
         arguments = json.loads(function_call["arguments"])
         return cls(**arguments)
+
+
+def openai_schema(cls):
+    if not issubclass(cls, BaseModel):
+        raise TypeError("Class must be a subclass of pydantic.BaseModel")
+
+    class Wrapper(cls):
```
Comment (jxnl) [inline openai_function_call/function_calls.py:118]: Should extend openaischema

**Review by jxnl — CHANGES_REQUESTED** [review]

**Comment by DaveOkpare** [comment]
@jxnl  Thanks for the tips. I have extended `OpenAISchema` and added `@wraps`

**Comment by jxnl** [comment]
Mind adding some texts to make sure it works.

**Pull request merged**